### PR TITLE
feat(actions): use aws ecr for openapi-generator image

### PR DESCRIPTION
**Pull request by mm-ivanovig**
### Description of the PR

use AWS ECR for openapi-generator image instead of DockerHub

### CODE REVIEW

File: .github/workflows/moneymeets-ci.yml [hunk 1]
```diff
@@ -6,21 +6,26 @@ jobs:
...
       - uses: actions/checkout@v2
 
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
       - name: Build image
         run: docker build -t $DOCKER_REGISTRY:$TAG .
```
Comment (marns93) [inline .github/workflows/moneymeets-ci.yml]:
```suggestion
      - run: docker build -t $DOCKER_REGISTRY:$TAG .
```

File: .github/workflows/moneymeets-ci.yml [hunk 2]
```diff
@@ -6,21 +6,26 @@ jobs:
     timeout-minutes: 15
 
     env:
-      DOCKER_REGISTRY: moneymeets/openapi-generator
+      DOCKER_REGISTRY: public.ecr.aws/moneymeets/openapi-generator
       TAG: frontend
 
     steps:
       - uses: actions/checkout@v2
```
Comment (marns93) [inline .github/workflows/moneymeets-ci.yml]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/moneymeets-ci.yml [hunk 3]
```diff
@@ -6,21 +6,26 @@ jobs:
...
+      - id: login-ecr-public
         if: github.ref == 'refs/heads/moneymeets-typescript-angular'
-        uses: docker/login-action@v1
+        uses: aws-actions/amazon-ecr-login@v1
         with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
+          registry-type: public
 
       - name: Publish image
```
Comment (marns93) [inline .github/workflows/moneymeets-ci.yml]: I think the name can be removed as well

**Review by marns93 — CHANGES_REQUESTED** [review]
:+1: 
Your changes looks good to me. You can update the other parts of the workflow to fix the warning with Nodejs 12 and remove `name` key to follow our convention.
Thank you. :)

**Comment by marns93** [comment]
@mm-ivanovig Please push the manual created images from DockerHub to ECR as well:
https://github.com/search?q=org%3Amoneymeets+moneymeets%2Fopenapi-generator&type=code

**Comment by mm-ivanovig** [comment]
> @mm-ivanovig Please push the manual created images from DockerHub to ECR as well: https://github.com/search?q=org%3Amoneymeets+moneymeets%2Fopenapi-generator&type=code

Pushed latest tag
Also I think easier to push backend images manually (amd64/arm64) - because for arm we need to check how to build them automatically. And also we need to upload manifest to the ECR for multi-arch images. 

@marns93 
Does it make sense to create a separate ticket for backend images upload automation?

**Comment by marns93** [comment]
> > @mm-ivanovig Please push the manual created images from DockerHub to ECR as well: https://github.com/search?q=org%3Amoneymeets+moneymeets%2Fopenapi-generator&type=code
> 
> Pushed latest tag Also I think easier to push backend images manually (amd64/arm64) - because for arm we need to check how to build them automatically. And also we need to upload manifest to the ECR for multi-arch images.

Thank you
> @marns93 Does it make sense to create a separate ticket for backend images upload automation?

Yes, I think we should do this.

**Review by marns93 — APPROVED** [review]
:+1:

**Comment by zyv** [comment]
@marns93 @mm-ivanovig 

https://buildjet.com/for-github-actions/docs/guides/arm

They support native ARM64 runners and claim to offer much faster hardware for cheaper price. Do you want to try it?

**Comment by marns93** [comment]
> @marns93 @mm-ivanovig
> 
> https://buildjet.com/for-github-actions/docs/guides/arm
> 
> They support native ARM64 runners and claim to offer much faster hardware for cheaper price. Do you want to try it?

Sure, we can do this. @mm-ivanovig Created a ticket https://moneymeets.atlassian.net/browse/MD-6711 yesterday and I added your link.

**Pull request merged**